### PR TITLE
EASY-1820 renamed licenseAgreement to depositAgreement (for JSon only yet)

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes EASY-1820 Change acceptLicenseAgreement to acceptDepositAgreement 

#### When applied it will
* have licenseAgreement renamed to depositAgreement in scala objects that are (de)serialized to json
* mention explicit versions of XSD so nothing will break when EASY-SCHEMA upgrades any
* validate the XML with schema's for files.xml and agreements.xml

#### How should this be manually tested?

* build and deploy together with ui
  * temporarily increment https://github.com/DANS-KNAW/easy-deposit-api/blob/8d3fa55954a21a7f29fcc170959413fea5f718b5/pom.xml#L28
  * as long as tests fail: `mvn clean install -DskipTests=true`
* [ ] check that the service runs: `sudo service easy-deposit-api status`
* go to `http://deasy.dans.knaw.nl/deposit` with a browser, login, create a deposit
* check the box at the bottom of the form to accept the depositor agreement
* save a draft

#### Related pull requests on github
* [ ] fix #89 , it does not start when deployed on http://develop.dans.knaw.nl/boxes/deasy-2018-11-30.box with  PR 267 merged into dtap
* must be merged/deployed together with https://github.com/DANS-KNAW/easy-deposit-ui/pull/113
* Update also [`application.properties`](https://github.com/DANS-KNAW/easy-validate-dans-bag/blob/076bd82cd95d365d2826bf932ba4e946f9cb5b0c/src/main/assembly/dist/cfg/application.properties#L11) of `easy-validate-dans-bag`

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.deposit/docs/AgreementsXml.scala [hunk 1]
```diff
@@ -20,21 +20,24 @@ import org.joda.time.DateTime
...
 
-object AgreementsXml {
+object AgreementsXml extends SchemedXml {
+  override protected val schemaNameSpace = "http://easy.dans.knaw.nl/schemas/bag/metadata/agreements"
+  override protected val schemaLocation = "http://easy.dans.knaw.nl/schemas/bag/metadata/agreements/2018/05/agreements.xsd"
+
   def apply(userId: String, dateSubmitted: DateTime, dm: DatasetMetadata): Try[Elem] = {
     for {
-      _ <- dm.licenceAccepted
+      _ <- dm.DepositAgreementAccepted
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/docs/AgreementsXml.scala]: with or without capital `D`?

File: src/test/scala/nl.knaw.dans.easy.deposit/docs/AgreementsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  }
+  "schema validation" should "succeed" in {
+    AgreementsXml(
+      "user",
+      DateTime.now,
+      DatasetMetadata().copy(
+        acceptDepositAgreement = true,
+        privacySensitiveDataPresent = PrivacySensitiveDataPresent.no
+      )
+    ).flatMap(AgreementsXml.validate) should matchPattern { case Success(_) => }
```
Comment (jo-pol) [inline src/test/scala/nl.knaw.dans.easy.deposit/docs/AgreementsSpec.scala]: @lindareijnhoudt 

why does this fail with

    SAXParseException; lineNumber: 1; columnNumber: 340; cvc-elt.1: Cannot find the declaration of element 'agr:agreements'

on
```xml
<agr:agreements xsi:schemaLocation="http://easy.dans.knaw.nl/schemas/bag/metadata/agreements http://easy.dans.knaw.nl/schemas/bag/metadata/agreements/2018/05/agreements.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:dcterms="http://purl.org/dc/terms/" xmlns:agr="http://easy.dans.knaw.nl/schemas/bag/metadata/agreements">
  <agr:licenseAgreement>
    <agr:depositorId>user</agr:depositorId>
    <dateAccepted>2018-03-22T21:43:01.000+01:00</dateAccepted>
    <agr:licenseAgreementAccepted>true</agr:licenseAgreementAccepted>
  </agr:licenseAgreement>
  <agr:personalDataStatement>
    <agr:signerId>user</agr:signerId>
    <agr:dateSigned>2018-03-22T21:43:01.000+01:00</agr:dateSigned>
    <agr:containsPrivacySensitiveData>false</agr:containsPrivacySensitiveData>
  </agr:personalDataStatement>
</agr:agreements>
```

File: src/test/scala/nl.knaw.dans.easy.deposit/docs/package.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import scala.util.Try
+import scala.xml.{ Elem, PrettyPrinter }
+
+package object docs extends DebugEnhancedLogging {
+  // pretty provides friendly trouble shooting for complex XML's
+  val prettyPrinter: PrettyPrinter = new scala.xml.PrettyPrinter(1024, 2)
+
+  implicit class validateWithSchema(val schemedXml: SchemedXml) extends AnyVal {
+    def validate(elem: Elem): Try[Unit] = {
+      assume(AgreementsXml.triedSchema.isSuccess)
```
Comment (jo-pol) [inline src/test/scala/nl.knaw.dans.easy.deposit/docs/package.scala]: only ignore in case of time out or no web access at all

**Comment by jo-pol** [comment]
replaced with #94

**Pull request closed without merging**